Source/WebCore/Modules/mediastream/MediaStreamTrackProcessor.cpp

```diff
@@ -98,21 +98,25 @@ void MediaStreamTrackProcessor::tryEnqueueingVideoFrame()
         m_readableStreamSource->enqueue(*videoFrame, *context);
 }
 
-Ref<MediaStreamTrackProcessor::VideoFrameObserverWrapper> MediaStreamTrackProcessor::VideoFrameObserverWrapper::create(ScriptExecutionContextIdentifier identifier, MediaStreamTrackProcessor& processor, Ref<RealtimeMediaSource>&& source)
+Ref<MediaStreamTrackProcessor::VideoFrameObserverWrapper> MediaStreamTrackProcessor::VideoFrameObserverWrapper::create(ScriptExecutionContextIdentifier identifier, MediaStreamTrackProcessor& processor, Ref<RealtimeMediaSource>&& source, unsigned short maxVideoFramesCount)
 {
     auto wrapper = adoptRef(*new VideoFrameObserverWrapper);
-    wrapper->initialize(identifier, processor, WTFMove(source));
+    wrapper->initialize(identifier, processor, WTFMove(source), maxVideoFramesCount);
     return wrapper;
 }
 
 MediaStreamTrackProcessor::VideoFrameObserverWrapper::VideoFrameObserverWrapper()
 {
 }
 
-void MediaStreamTrackProcessor::VideoFrameObserverWrapper::initialize(ScriptExecutionContextIdentifier identifier, MediaStreamTrackProcessor& processor, Ref<RealtimeMediaSource>&& source)
+void MediaStreamTrackProcessor::VideoFrameObserverWrapper::initialize(ScriptExecutionContextIdentifier identifier, MediaStreamTrackProcessor& processor, Ref<RealtimeMediaSource>&& source, unsigned short maxVideoFramesCount)
 {
-    callOnMainThread([protectedThis = Ref { *this }, identifier, processor = WeakPtr { processor }, source = WTFMove(source)] () mutable {
-        protectedThis->m_observer = makeUnique<VideoFrameObserver>(identifier, WTFMove(processor), WTFMove(source));
+    callOnMainThread([protectedThis = Ref { *this }, identifier, processor = WeakPtr { processor }, source = WTFMove(source), maxVideoFramesCount] () mutable {
+#if PLATFORM(COCOA)
+        if (source->deviceType() == CaptureDevice::DeviceType::Camera)
+            maxVideoFramesCount = 1;
```

Comment:
We should test to make sure this isn't also necessary for screen/window capture.

Reply:
This is on iOS only.
On macOS, camera and screen/window capture is fine.